src/input-states.js

```diff
@@ -0,0 +1,215 @@
+// input-states.js
+
+var _ = require('underscore');
+
+/**
+ * Getter and Setter for InPort states.
+ *
+ * @this is not used
+ * @param node a node facade
+ * @param vnid an identifier that distinguishes the set of VNI states
+ * @param portName name of the input port whose state is being recorded
+ * @param socketId index of the socket or undefined
+ * @param state a State object containing both a data payload and an lm.
+ *
+ * @return getter returns the current input port state state.
+ *         setter returns the current context
+ *
+ * Usage to retrieve state:
+ *  inputState(node, vnid) : {portName: currentState, multiPortName: [currentState]}
+ *  inputState(node, vnid, portName, socketId) : currentState
+ * Usage to store state:
+ *  inputState(node, vnid, {portName: currentState, multiPortName: [currentState]}) : this
+ *  inputState(node, vnid, portName, socketId, state) : this
+ */ 
+module.exports = function(node, vnid, portName, socketId, state) {
+    if (!node.inPorts || !node.outPorts) throw Error("This isn't a Component node");
```

Comment:
Can you quote the style guide? I don't see anything about the use of optional braces.